src/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/HttpConnection.cs

```diff
@@ -269,7 +269,8 @@ private void ConsumeFromRemainingBuffer(int bytesToConsume)
             else
             {
                 Debug.Assert(_pool.UsingProxy);
-                await WriteAsciiStringAsync(uri.IdnHost).ConfigureAwait(false);
+                await WriteAsciiStringAsync(uri.HostNameType == UriHostNameType.IPv6 ?
+                    "[" + uri.IdnHost + "]" : uri.IdnHost).ConfigureAwait(false);
```

Comment:
How about string interpolation instead of string concatenation manually i.e. $`"[{uri.IdnHost}]" `

Reply:
String interpolation would be more expensive here.

Reply:
Does string interpolation has significant performance difference than string concatenation. I believe, string interpolation has better readability.

Reply:
In this case it'll compile down to a call to string.Concat:
```C#
string.Concat("[", uri.IdnHost, "]");
```
which will incur just a single allocation for the destination string, with the data from each constituent string blitted into it.  In contrast, the string interpolation would compile down to the equivalent of:
```C#
string.Format("[{0}]", new object[] { uri.IdnHost });
```
and would thus incur at least the destination string and an object array allocation, plus the work that has to be done inside of Format, which involves a StringBuilder.

Reply:
Another useful note on string.Concat vs. string.Format https://github.com/dotnet/corefx/pull/8025#discussion_r60852930 (I want "bookmark favorite comment" feature in GitHub) :octocat:

Reply:
@stephentoub  : Thank you for explaining it with implementation side. @kasper3  : Thank you for commenting with interesting thread.

Reply:
Since we're now using uri.Host in the GetConnectionKey path, seems like it would make sense to use it here as well for consistency and to avoid the extra alloc (maybe?).

Reply:
I agree, will change and add comment as well.